Fix logic for setting bulb attributes when bulb is OFF

sdrapha opened the pull request:
Fix logic for setting bulb attributes when bulb is OFF
also add Tests

Fix issue #184

sdrapha committed:
Fix code for setting bulb attributes when bulb OFF
also add Tests

webdjoe commented:
Thank you, @sdrapha. Love the tests! I'll include it in a new release in a week or so

sdrapha committed:
Fix tox.ini so it runs in different systems
I had systems with python 3.11, python 3.10 and python 3.8.
none of them with 3.9.
Add all of them to the list so we can properly test the package.
Remove 'basepython' line to fix the pylint, flake8 and mypy tests

sdrapha committed:
Change timeout value.
The cloud api will respond in 7 seconds if the device is unreacheable.
Set this package timeout to 8 so there is enough time to get that response.
Previously value was set to 5. which was not enough.
Above all. the main reason for the change is that the V2 cloud api endpoint may often take around 4 to 6 seconds to respond back.
So the previous value was errounesly setting devices as offline when they were not.
